data/types/base.go

```diff
@@ -283,6 +283,18 @@ func (d Datum) ToStringArray(fieldName string, errs validate.ErrorProcessing) *[
 	return &stringArray
 }
 
+func (d Datum) ToMap(fieldName string, errs validate.ErrorProcessing) *map[string]interface{} {
```

Comment:
I don't think you need to return a pointer to a map here. You should be able to return nil for a map (or array or interface).